Add Cypress Accessibility Branch Review documentation (#6110)
cypress-io/cypress-documentation: docs/accessibility/core-concepts/comparing-reports.mdx

---
title: 'Comparing Accessibility Reports | Cypress Documentation'
description: 'Review the main areas to pay attention to when first reviewing an accessibility report for a Cypress run.'
sidebar_position: 40
sidebar_label: Compare reports
---

<ProductHeading product="accessibility" />

# Compare reports

Accessibility reports from different runs can be compared in the [Branch Review](/cloud/features/branch-review#Getting-Started) area of Cypress Cloud. This allows you to instantly see if any new issues have been introduced, and drill in to see full-page HTML and CSS snapshots showing only the new issues.

See the video below for an example of using Branch Review in a Pull Request:

<DocsVideo

>   src="https://www.youtube.com/embed/k3vq57O4P4M"

Reviewer: This video is great!

  title="Accessibility branch review demo"
/>

## Use cases

Comparing the results from different runs is useful in multiple scenarios.

**Key use cases:**

> - **Pre-merge checks**: Know if any net-new issues are introduce by UI code changes.

Reviewer: ```suggestion
- **Pre-merge checks**: Know if any net-new issues are introduced by UI code changes.
```

- **Monitoring changes**: Compare nightly monitoring runs and track down the introduction of new problems caused by underlying changes in the application.
- **Detecting content issues**: Sometimes content editors can introduce accessibility issues unrelated to code changes. Seeing the example issues presented visually, in context, helps you quickly triage whether you are dealing with a recent code change issue, or a content authorship problem.
- **Reviewing AI-generated code changes**: The increased use of AI to generate and/or review front-end code creates some increased risks of accessibility regressions making it to production. The increase or decrease of accessibility issues when reviewing a pull request helps you understand the impact of the change.

> - **Tracing the introduction of issues**: With dropdowns for each run, it's easy to rapidly compare different A and B runs to find the exact commit that introduced a problem.

Reviewer: @marktnoonan I would also add the use case for using branch review is to show that you have RESOLVED an issue that your code changes were looking to resolve - and can easier share that overview with your team to more quickly review code changes.

Author: updated

## Content of the report

<DocsImage
  src="/img/accessibility/core-concepts/branch-review-intro.png"
  alt="Cypress Accessibility Branch Review with arrows calling out the sections for new failed rules, resolved rules, and failed rules with changes. New failed rules are defined as rules that just started failing on the new branch. Resolved rules are defined as rules that got fully fixed in the new branch. Failed rules with changes are defined as rules that failed on both branches, but with different element counts or different related pages."
  noBorder={true}
/>

The Branch Review report is prioritized into three sections:

- **New failed rules**: Rules that were passing every time they ran on the `base` run but now have failures in the `changed` run.
- **Resolved rules**: Rules that had some failures detected in the `base` run but have no failures on the `changed` run. This helps you celebrate the wins and recognize when a new run has gone green.
- **Failed rules with changes**: Rules that were failing in both runs, where the elements with failures detected have increased, decreased, or changed in some way.

> This organization of the report brings the most significant results to the top - regressions of rules that had been fully passing in the `base` run. Increases or decreased in the element counts for rules that already have some failures provide a good sense of progress, but can tend to be noisier, especially if the runs have different content or other conditions that changed in between.

Reviewer: ```suggestion
This organization of the report brings the most significant results to the top - regressions of rules that had been fully passing in the `base` run. Increases or decreases in the element counts for rules that already have some failures provide a good sense of progress, but can tend to be noisier, especially if the runs have different content or other conditions have changed in between.
```

## How to compare runs

> The first step is to get to the Branch Review area of Cypress Cloud, which will let you compare one branch against another - or different runs on the same branch, if needed. We refer to be baseline fun for comparison as the `base` run, and the changes we are comparing with as the `changed` run.

Reviewer: ```suggestion
The first step is to get to the Branch Review area of Cypress Cloud, which will let you compare one branch against another - or different runs on the same branch, if needed. We refer to the baseline run for comparison as the `base` run, and the changes we are comparing as the `changed` run.
```

<DocsImage
  src="/img/accessibility/core-concepts/branch-review-run-card.png"
  alt="A couple of cards showing various details about Cypress test runs, including pass/fail counts and metadata. An arrow points to the branch name."
/>

### From the project list

<DocsImage
  src="/img/accessibility/core-concepts/branch-review-project-card.png"
  width="40%"
  alt="A card showing the name of an example project and related branches to choose."
/>

The project card shows three active branches for the project. You can click through to any of these to enter Branch Review with the newest run on that branch pre-selected as the `changed` run.

### From the main navigation

<DocsImage
  src="/img/accessibility/core-concepts/branch-review-side-nav.png"
  width="40%"
  alt="The main Cypress Cloud navigation showing Latest Runs, Branches, and Analytics menu items. An arrow points to Branches."
/>

When inside of a project, you can select "Branches" in the main navigation to see a full, filterable list of available branches and choose one to set as the `changed` run.

## FAQ

> ### How to I ensure a good comparison?

Reviewer: ```suggestion
### How do I ensure a good comparison?
```

> The best subjects to compare are passing runs that ran similar tests on the same set of content. This means that each run visited roughly the same pages and completed the same kinds of workflows. In this situation, any diff in the results is likely the result of changes present in the newer run. This is usually what happens out-of-the box when comparing a pull-request branch with your main brach. Test failures or errors on either run

Reviewer: This sentence is unfinished at the end of this paragraph.

> That said, it still possible and valid to compare runs from different points in time with different sets of test results, as long as you bear in mind all the potential sources of difference between the two runs, which you can evaluate for yourself as you explore the results.

Reviewer: ```suggestion
That said, it is still possible and valid to compare runs from different points in time with different sets of test results, as long as you bear in mind all the potential sources of differences between the two runs, which you can evaluate for yourself as you explore the results.
```

### What is the purpose of the Beta label?

This indicates the feature is ready for use and actively seeking feedback based on real usage of the current implementation. We have a few known issues to work through on our side before we consider this fully production-ready and remove the beta label. These issues only affect a subset of projects -- in most cases everything is working as intended. If you see anything unexpected, please hit the feedback button and let us know.

### Why do I see some views (pages or components) changing from run-to-run?

> URLs with dynamic slugs in them can appear as "new" pages in some situations. This behavior can be adjusted with [View configuration] to make sure page names match across runs by wildcarding parts of the URL as needed.

Reviewer: Link missing to view configuration here?
